### DISCUSSION TITLE
walletConnect appkit

### DISCUSSION THREAD

**Discussion by lorof** [wevm/wagmi#4133] [Q&A]
While working with the WalletConnect connector from the wagmi repo, we encountered numerous issues that seem to depend on the @walletconnect/modal and @walletconnect/ethereum-provider packages. After reaching out to WalletConnect support, they recommended using https://docs.walletconnect.com/appkit/upgrade?platform=react. However, this guide suggests removing all current connectors and adopting their modal. My question is: is it possible to retain our current custom modal with multiInjectedProviderDiscovery and still use the new WalletConnect modal? Will there be an update to the connectors in wagmi for this AppKit v5?

**Comment by belimposition** [comment]
also interested - I'm doing a similar task

**Comment by mastyf** [comment]
Also interested

**Comment by skotancilar** [comment]
I’m looking forward to the answer as I’m working on a similar task.
